drf_spectacular/openapi.py

```diff
@@ -468,13 +468,17 @@ def _map_serializer_field(self, field, direction):
             schema = build_array_type({})
             # TODO check this
             if not isinstance(field.child, _UnvalidatedField):
-                map_field = self._map_serializer_field(field.child, direction)
-                items = {
-                    "type": map_field.get('type')
-                }
-                if 'format' in map_field:
-                    items['format'] = map_field.get('format')
-                schema['items'] = items
+                if is_serializer(field.child):
+                    component = self.resolve_serializer(field.child, direction)
+                    return append_meta(build_array_type(component.ref), meta) if component else None
+                else:
+                    map_field = self._map_serializer_field(field.child, direction)
```

Comment:
i like it, but i think you could have even gone further! fyi, this is one of the last pieces of legacy code. that is why there was a TODO there (which you can rightfully remove now).

since `ListField` is functionally almost equal to `ListSerializer`, i think it makes sense to also handle them (almost) equally.

```python
         if isinstance(field, serializers.ListField):
            if isinstance(field.child, _UnvalidatedField):
                return append_meta(build_array_type({}), meta)
            elif is_serializer(field.child):
                component = self.resolve_serializer(field.child, direction)
                return append_meta(build_array_type(component.ref), meta) if component else None
            else:
                schema = self._map_serializer_field(field.child, direction)
                return append_meta(build_array_type(schema), meta)
```

fwiw we maybe could have even merged those two together, but i'm not sure about the `_UnvalidatedField` in the `ListSerializer` case.